### ISSUE TITLE
Fix `named-grid-areas-no-invalid` false positives for mix of tabs and spaces

### ISSUE THREAD

**Issue by EyedPeas** [stylelint/stylelint#8134]
### What minimal example or steps are needed to reproduce the bug?

``` css
.test-grid {
	grid-template:
		"a a a"     /* 3 whitspaces */
		"b b	b"; /* 2 whitespaces 1 tab */
}
```

### What minimal configuration is needed to reproduce the bug?

```json
"rules": {
  "named-grid-areas-no-invalid": true
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AXGBnFBaA5gJwEsoACYAHQDsyV8is0BbABwBsBDNeS6nkN4-mzIhuKXgCNi46tJABuSgF9uFEABoQAMwIsYAOTYM4iGAA9DrGEjDp068BAracCEOQrFiws2gpR0wvDEANqUnp7CmACeuiwEFNiQTgQ4WJhsfmx4UMJhxAC6SvZJzgBiEHgMHK4AVuiO9rBMdojuESDRsfFiCF4g7GiYwmp5kSgxMHEJWCUpaSgZUFk5vcIDGD1FikA)

### Which Stylelint-related dependencies are you using?

```json
{		
  "stylelint": "^15.10.3",
  "stylelint-config-standard-scss": "^11.0.0"
}

```

### What did you expect to happen?

No error "Expected same number of cell tokens in each string  (named-grid-areas-no-invalid)"
=> ignoring tabs

### What actually happened?

Error "Expected same number of cell tokens in each string  (named-grid-areas-no-invalid)"  was reported

### Do you have a proposal to fix the bug?

ignore tabs

**Comment by romainmenke** [comment]
is `\` a trash token?

No, it is not a trash token and this is indeed preprocessed.
Or rather there is double tokenization step.

The tokenizing algorithm described in https://drafts.csswg.org/css-grid/#valdef-grid-template-areas-string is expected to be applied the "normalized" value of the string and not the raw value as it appears in source.

The tokenizer algorithm will see: `"b b b"`, as if the newline was never there.

------

I was curious and implemented this algorithm in `css-tokenizer`.
It's a good fit to include it there.

But it does reveal some interesting issues.
Because there is a double tokenization step the source indexes don't line up when there are escape sequences or other kinds of values that are normalized. So it is tricky to do fine grained reporting. Indicating that there is an issue somewhere in the string is trivial. But indicating which cell is incorrect is not.

I did expect this rule to also give feedback on trash tokens, but that doesn't seem to be currently covered.
Should we open an issue for adding that?

**Comment by Mouvedia** [comment]
> No, it is not a trash token and this is indeed preprocessed.

Thanks for the confirmation that's what I would expect.

> So it is tricky to do fine grained reporting. Indicating that there is an issue somewhere in the string is trivial. But indicating which cell is incorrect is not.

Would reporting the problematic string(s) be easier?

> I did expect this rule to also give feedback on trash tokens, but that doesn't seem to be currently covered.
Should we open an issue for adding that?

Let's separate the issues but one PR may fix them both.
i.e. with 2 changesets
